homeassistant/components/opensprinkler/sensor.py

```diff
@@ -0,0 +1,182 @@
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_TIMESTAMP
+
+    @property
+    def icon(self) -> str:
+        """Return icon."""
+        return "mdi:history"
+
+    @property
+    def name(self) -> str:
+        """Return the name of this sensor including the device name."""
+        return f"{self._name} Last Run"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return f"{self._entry_id}_{self._entity_type}_{self._name}_last_run"
+
+    def _get_state(self):
+        """Retrieve latest state."""
+        last_run = self._device.device.last_run
+        utc_time = datetime.fromtimestamp(last_run, UTC_TZ)
+        return utc_time.isoformat()
+
+
+class RainDelayStopTimeSensor(OpensprinklerSensor, Entity):
+    """Represent a sensor that for rain delay stop time."""
+
+    def __init__(self, entry_id, name, device, coordinator):
+        """Set up a new opensprinkler rain delay stop time sensor."""
+        self._entry_id = entry_id
+        self._name = name
+        self._device = device
+        self._entity_type = "sensor"
+        super().__init__(coordinator)
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_TIMESTAMP
+
+    @property
+    def icon(self) -> str:
+        """Return icon."""
+        return "mdi:weather-rainy"
+
+    @property
+    def name(self) -> str:
+        """Return the name of this sensor including the device name."""
+        return f"{self._name} Rain Delay Stop Time"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return f"{self._entry_id}_{self._entity_type}_{self._name}_rdst"
+
+    def _get_state(self):
+        """Retrieve latest state."""
+        rdst = self._device.device.rain_delay_stop_time
+        if rdst == 0:
+            return "Not in effect"
+
+        utc_time = datetime.fromtimestamp(rdst, UTC_TZ)
+        return utc_time.isoformat()
+
+
+class StationSensor(OpensprinklerSensor, Entity):
+    """Represent a sensor for status of station."""
+
+    def __init__(self, entry_id, station, device, coordinator):
+        """Set up a new opensprinkler device sensor."""
+        self._entry_id = entry_id
+        self._station = station
+        self._device = device
+        self._entity_type = "sensor"
+        super().__init__(coordinator)
+
+    @property
+    def name(self) -> str:
+        """Return the name of this sensor."""
+        return self._station.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return f"{self._entry_id}_{self._entity_type}_station_{self._station.name}"
+
+    def _get_state(self) -> bool:
+        """Retrieve latest state."""
+        return self._station.status
```

Comment:
What values can the status be?

Reply:
`manual`, `once_program`, `idle`, `program`,`waiting`